app/controllers/admin/tag_sets_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+##
+# This class is the controller for Tag Sets which are used to link together two related tag groups.
+# It allows you to create and view Tag Sets.
+class Admin::TagSetsController < ApplicationController
+  authorize_resource
+
+  def index
+    @tag_sets = TagSet.all
+
+    respond_to { |format| format.html }
+  end
+
+  def show
+    @tag_set = TagSet.find(params[:id])
+
+    respond_to { |format| format.html }
+  end
+
+  def new
+    @tag_set = TagSet.new
+
+    respond_to { |format| format.html }
+  end
+
+  def create
+    @tag_set = TagSet.new(tag_set_params)
+
+    respond_to do |format|
+      if @tag_set.save
+        flash[:notice] = 'Tag Set was successfully created.'
+        format.html { redirect_to admin_tag_set_path(@tag_set) }
+      else
+        format.html { render action: :new }
+      end
```

Comment:
We are not returning feedback on error. It seems to just go to new TagSet page again. Does it show a banner?

Reply:
Do we need to add anything to errors collection explicitly or does it just add it automatically?

Reply:
Right, save will run validations. They will add the errors. I think it is OK then.